enchant/_enchant.py

```diff
@@ -109,6 +109,13 @@ def from_system() -> Optional[str]:
         "libenchant-1",
     ]
 
+    # On M1 macs, Homebrew is in /opt, which isn't on the default
+    # DYLD_LIBRARY_PATH. Look for the homebrew path explicitly.
+    if sys.platform == "darwin" and platform.machine() == "arm64":
+        candidates.extend(
+            [f"/opt/homebrew/lib/{candidate}" for candidate in candidates]
```

Comment:
Nit: You can remove the surrounding `[…]` here to convert the list-comprehension into an iterator, which `list.extend()` can handle as well. It's a micro-optimization as using an iterator is more efficient than first building the complete list internally first, just to have each element appended to `candidates` before that list is deleted. More important than that optimization is the code-style: `flake8` and `ruff` will also give you that advise.

Sadly there are still no Apple Silicon runners available via GitHub actions: https://github.com/github/roadmap/issues/528 is now schedules for 2023-Q3.

Otherwise looks good and thank you for looking into this.

Reply:
I'm very much aware of the list-comprehension optimisation. However, in this case, it's not possible (and, FWIW, flake8 doesn't suggest it).

It doesn't work because candidates is being extended by iterating over itself. If you use the list comprehension form, the list that is being iterated over is effectively infinite, because every time a new item is added to the list, the list gets longer, which adds another item to the list, and so on. Using the list-literal form means a copy of the list is created statically to pass into `extend()`, which avoids the infinite loop.

(I originally pushed the change without thinking about it, but then realised why it wouldn't work, so I've reverted it; however, this has unfortunately reset the CI status).

Reply:
Yes, you're right. Thank you for looking into this and doing all the work.

Reply:
@pmhahn No problems at all. 

What's the project policy around making new releases? I notice there hasn't been a published release for over 18 months, but it also looks like most of the changes in that time appear to have been documentation, CI config updates, or Black formatting. What sort of timeframe should I anticipate for this change being incorporated in a formal release?

Reply:
I'm new to the project myself and as such have not yet participated in doing a new release: I have no idea myself.